addOns/oast/src/main/java/org/zaproxy/addon/oast/OastInsertPayloadMenu.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast;
+
+import java.awt.Component;
+import java.util.Map;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import javax.swing.text.JTextComponent;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.ExtensionPopupMenu;
+
+@SuppressWarnings("serial")
+public class OastInsertPayloadMenu extends ExtensionPopupMenu {
+    private static final long serialVersionUID = 2L;
+    private JTextComponent lastInvoker;
+    private ExtensionOast extension;
+
+    public OastInsertPayloadMenu(ExtensionOast extension) {
+        super();
+        this.extension = extension;
+        setText(Constant.messages.getString("oast.popup.menu.insertPayload"));
+        add(new OastInsertPayloadHandler(this, true));
+        add(new OastInsertPayloadHandler(this, false));
+    }
+
+    private class OastInsertPayloadHandler extends JMenu {
+        private OastInsertPayloadMenu parentMenu;
+        private boolean createNewPayload;
+
+        public OastInsertPayloadHandler(
+                OastInsertPayloadMenu parentMenu, boolean createNewPayload) {
+            super(
+                    Constant.messages.getString(
+                            "oast.popup.menu." + (createNewPayload ? "new" : "existing")));
+            this.parentMenu = parentMenu;
+            this.createNewPayload = createNewPayload;
+            initialize();
+        }
+
+        public void initialize() {
+            Map<String, OastService> services = extension.getOastServices();
+            for (Map.Entry<String, OastService> serviceEntry : services.entrySet()) {
+                String name = serviceEntry.getKey();
+                OastService service = serviceEntry.getValue();
+                JMenuItem menuItem = new JMenuItem(name);
+                this.add(menuItem);
+                menuItem.addActionListener(
+                        e -> {
+                            try {
+                                parentMenu
+                                        .getLastInvoker()
+                                        .replaceSelection(service.getPayload(createNewPayload));
+                            } catch (Exception e1) {
+                                e1.printStackTrace();
```

Comment:
Generally exceptions should be logged not printed. But in this case the user should be informed as well.

Reply:
This should show a proper error message to the user like e.g. `Failed to generate an OAST payload.\n…`.
For the log I'd go with a `warn` rather than `error`, the latter we use for likely bugs.

Reply:
Bump.

Reply:
currently it display the following error message to the user : 
![Screenshot from 2023-03-02 13-23-37](https://user-images.githubusercontent.com/87857234/222366012-e95fd4fa-7221-42d2-8ec1-9b3980787c6a.png)

Do you want it to be the following way ? 
![Screenshot from 2023-03-02 13-24-24](https://user-images.githubusercontent.com/87857234/222366084-9923dabf-13ce-427f-933f-200326ab1971.png)

Reply:
Yes, reading it again I think we should be more clear, e.g. `Failed to generate and insert the OAST payload.\n…`.

Reply:
okk sure